tests/python/pants_test/backend/python/rules/test_resolve_requirements.py

```diff
@@ -0,0 +1,52 @@
+# coding=utf-8
```

Comment:
I think that this test skeleton makes sense. Ultimately, what we care about is that we can translate our parameters into a meaningful PEX created in the real world, e.g. that we are properly calling PEX's CLI flags. Rather than having to actually run the built PEX, we can simply inspect the `PEX-INFO` to make sure it's all set up properly.

I still have to actually implement part of these tests, but let me know if this looks like the right direction.

cc @stuhood

Reply:
For that case, you might consider using a unit test as described in https://github.com/pantsbuild/pants/pull/7679#pullrequestreview-235267227 (ie, via `run_rule`), and validating that you see the appropriate args being passed to pex as an `ExecuteProcessRequest`.